src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -99,4 +101,15 @@ public Book updateBook(@RequestBody Book book, @PathVariable Long id)
         bookRepository.findById(id).orElseThrow(BookNotFoundException::new);
         return bookRepository.save(book);
     }
+
+    @GetMapping("/isbn/{isbn}")
+    @ApiOperation(value = "Finds a book on the open library's listing by ISBN")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "When a book exists"),
+        @ApiResponse(code = 404, message = "When a book could not be found")
+    })
+    public OpenLibraryBook findBookByIsbn(@PathVariable String isbn) throws BookNotFoundException {
+        OpenLibraryService service = new OpenLibraryService();
```

Comment:
Remember that the description of the card says you should first search in the bookRepository, and only then you should search in the open library service if it wasn't present in the repository